pkg/scheduler/api/node_info.go

```diff
@@ -124,6 +141,12 @@ func (ni *NodeInfo) AddTask(task *TaskInfo) error {
 			ni.Idle.Sub(ti.Resreq)
 		}
 
+		// removing pod resources not maintained by kube-batch from node.ALlocatable
+		if ti.Job == "" && ti.Pod.Status.Phase == v1.PodRunning {
```

Comment:
Why change allocatable? It's reported from `Node`, not related to pod.

Reply:
because allocatable contains resource already used by other pods not managed by kube-batch. they are literally not "allocable" by kube-batch.

Reply:
we need to include all running resource in kube-batch.

Reply:
Before we add backfill capability in this PR, kube-batch makes scheduling decisions by comparing a task's request resource amount against a node's *idle* resource amount. It does not use a node's *allocatable* resource amount. Kube-batch will be happy as long as the node's idle resource amount is correct. 

With the backfill capability added in this PR, kube-batch not only needs to check a node's idle resource amount, but also needs to check a node's *allocatable* amount.  For example, kube-batch may be able to backfill a task if the task's request resource amount is greater than a node's idle resource amount but less than a node's allocatable resource amount.  Therefore, kube-batch has to make sure both *idle* amount and *allocatable* amount are accurate. 

In a cluster where kube-batch scheduler runs side by side with other schedulers, a node may have multiple running Pods among which some are managed by kube-batch and some are managed by other schedulers.  Kube-batch can only preempt pods managed by itself. It has no right to preempt pods managed by other schedulers. In this case, we'd better deduct a node's allocatable resource amount when we see a running pod which does not belong to kube-batch.

Reply:
> Kube-batch can only preempt pods managed by itself. It has no right to preempt pods managed by other schedulers. 

In this case, we need to mark it as un-preemptable in conformance plugin.
As we discussed before, we should not change the meaning of this field.

Reply:
> > Kube-batch can only preempt pods managed by itself. It has no right to preempt pods managed by other schedulers.
> 
> In this case, we need to mark it as un-preemptable in conformance plugin.
> As we discussed before, we should not change the meaning of this field.

we are not changing the meaning of the this field. we are FIXING what's wrong in the first place. 

Explain to me please why you think it's "correct" that resources accounted for as "allocable" is not actually allocable for the scheduler.

Reply:
@k82cn Imagine kube-batch gets a starving task, and it plans to reserve all remaining idle resources on node X for the starving task. But before reserving the resources on node X, kube-batch has to make sure the allocatable resources (i.e. the total resources) of node X is bigger enough to accommodate the task. 

According to the current implementation, if node X has 5G memory and 4G memory are taken by jobs that not managed by kube-batch scheduler, node X's allocatable resource is still 5G. In this case, kube-batch may think node X has enough allocatable resource to accommodate the starving task. But the reality is different. 4G memory is occupied by tasks on which kube-batch has no control. Kube-batch cannot evict those tasks to make room for the starving task.

Reply:
> Explain to me please why you think it's "correct" that resources accounted for as "allocable" is not actually allocable for the scheduler.

It means allocatable resources on the node, not the scheduler. If we want to trace the "allocatable" resource in scheduler, please introduce a new field for it.

> In this case, kube-batch may think node X has enough allocatable resource to accommodate the starving task. 

Not, we should consider 1G memory for kube-batch; that's why `Idle` there.

Reply:
For starving tasks, we do not just compare a task's requesting resource with idle resource, we also compare a task's requesting resource against the total resource on the node.

Reply:
"It means allocatable resources on the node"

if it's not allocatable for the scheduler, how is it even allocatable on the node?